examples/llm/inference/benchmarking/yamls/7b.yaml

```diff
@@ -0,0 +1,39 @@
+benchmark_name: MosaicDeepspeed_1xA100-40GB-bf16-GPT-7B
+
+max_seq_len: 2048
+tokenizer_name: gpt2
+
+# Tokenizer
+tokenizer:
+  name: ${tokenizer_name}
+  kwargs:
+    model_max_length: ${max_seq_len}
+  non_eos_token_id: 17
+
+model:
+  name: mosaic_gpt
+  init_device: cpu
+  tokenizer_name: ${tokenizer_name}
+  d_model: 4096
+  n_heads: 32
+  n_layers: 32
+  mlp_ratio: 4
+  max_seq_len: ${max_seq_len}
+  vocab_size: 50368
+  init_std: 0.02
+  attn_pdrop: 0.0
+  resid_pdrop: 0.0
+  emb_pdrop: 0.0
+  attn_impl: triton
+
+precision: amp_bf16
+
+batch_sizes: [1, 2, 4, 8, 16, 32, 64]
+input_lengths: [128]
+output_lengths: [8]
+num_runs: 5
+
+num_warmup_batches: 3
+
+use_precision_context: False
+use_deepspeed: True
```

Comment:
I think it would make sense to rename  the `precision` and `use_precision_context` flags.

Perhaps like:

```
param_dtype: [fp32, bf16, fp16]
autocast: [none, amp_bf16, amp_fp16]
```

Right now, the setups I would like to test are not easy to configure:
* bf16 weights, no autocast
* bf16 weights, autocast amp_bf16 (which will upcast certain ops to fp32)
* fp32 weights, autocast amp_bf16 (expected behavior is identical to above)

Leaving the naming and stuff to you, but whatever convention you decide I will update `hf_generate.py` to match.